datalad/distributed/drop.py

```diff
@@ -415,7 +415,9 @@ def _drop_dataset(ds, paths, what, reckless, recursive, recursion_limit, jobs):
                 res['message'] = message
             error_messages = r.get('error-messages')
             if error_messages:
-                res['error_messages'] = error_messages
+                res['error_message'] = '\n'.join(
```

Comment:
Not sure what you mean by `+=` - to put both the original `error_messages` list and the `error_message` string in the `res` dictionary? Or to append the string to `res['error_message`]? The latter wouldn't matter, because `res` is instantiated with a `dict(...)` moments earlier.

Reply:
I meant to append, but you are right - here it doesn't matter. However, `r` could have an `error_message` already that we ignore here. That may not currently being the case, but that can change whenever things change at a lower-level function.

Reply:
looking at the code, indeed the `error_message` shouldn't end up being there.  And indeed it might ;)  if there is an assumption that it is not there:

```suggestion
                assert 'error_message' not in res
                res['error_message'] = '\n'.join(
```
to ensure that, so that if code changes, tests (if case is covered) would alert us that assumption is broken

Reply:
This is esssentially saying that the following code:

```py
d = dict(a=5)
# <imagine 7 lines of code with no reference to `d`>
d['b'] = 7
```

should become

```py
d = dict(a=5)
# <imagine 7 lines of code with no reference to `d`>
assert 'b' not in d
d['b'] = 7
```

if this is to become a standard approach, we have to add a few thousand patch to the code base ASAP ;-)